product_default_image/models/product_template.py

```diff
@@ -0,0 +1,299 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from collections import defaultdict
+
+from odoo import api, fields, models, tools
+
+
+class ProductTemplate(models.Model):
+
+    _inherit = 'product.template'
+
+    auto_change_image = fields.Boolean(
+        string='Auto Change Image',
+        default=True,
+        help='Allow/disallow automatic changes to product image. '
+             'Uncheck to stop being changed to a default image.',
+    )
+    img_type = fields.Selection(
+        string='Image Type',
+        selection=[
+            ('default_global', 'Global'),
+            ('default_category', 'Category'),
+            ('no_image', 'No Image'),
+            ('custom', 'Custom'),
+        ],
+        required=True,
+        readonly=True,
+        default='no_image',
+    )
+
+    @api.model
+    def default_get(self, vals):
+
+        vals = super(ProductTemplate, self).default_get(vals)
+
+        company = self.env.user.company_id
+        target = company.product_image_target
+
+        if target in ['category', 'global_category'] and vals.get('categ_id'):
+            categ = self.env['product.category'].browse(vals['categ_id'])
+            vals.update({
+                'image': categ.image,
+                'img_type': 'default_category',
+            })
+
+        if target in ['global', 'global_category'] and not vals.get('image'):
+            vals.update({
+                'image': company.product_image,
+                'img_type': 'default_global',
+            })
+
+        tools.image_resize_images(vals)
+        return vals
+
+    @api.model
+    def search_change_images(self, from_types, to_type, to_img_bg=None,
```

Comment:
For the doc string, for the args could I point to the methods below that this method calls? (those ones have docstrings.). Or should I restate all that stuff here?